spec/service_worker/index.html

```diff
@@ -3589,6 +3632,148 @@ <h2 id="status">Status of This Document</h2>
     </spec-algorithm>
   </spec-section>
 
+  <spec-section id="foreign-fetch-scope-match-algorithm">
+    <h1>Match Service Worker for Foreign Fetch</h1>
+
+    <spec-algorithm>
+    <dl>
+      <dt>Input</dt>
+        <dd><var>requestURL</var>, a <a href="https://url.spec.whatwg.org/#concept-url">URL</a></dd>
+      <dt>Output</dt>
+        <dd><var>worker</var>, a <a href="#dfn-service-worker">service worker</a></dd>
+    </dl>
+    <ol>
+      <li>Run the following steps atomically.</li>
+      <li>Let <var>registration</var> be the result of running the <a href="#scope-match-algorithm">Match Service Worker Registration</a> algorithm passing <var>requestURL</var> as the argument.</li>
+      <li>If <var>registration</var> is null, return null.</li>
+      <li>Let <var>worker</var> be <var>registration</var>'s <a href="#dfn-active-worker">active worker</a>.</li>
+      <li>If <var>worker</var> is null, return null.</li>
+      <li>Let <var>requestURLString</var> be the <a href="https://url.spec.whatwg.org/#concept-url-serializer">serialized</a> <var>requestURL</var>.</li>
+      <li>For each <a href="https://url.spec.whatwg.org/#concept-url">URL</a> <var>scope</var> in <var>worker</var>'s <a href="#dfn-foreign-fetch-scopes">list of foreign fetch scopes</a>:
+        <ol>
+          <li>Let <var>scopeString</var> be the <a href="https://url.spec.whatwg.org/#concept-url-serializer">serialized</a> <var>scope</var>.</li>
+          <li>If <var>requestString</var> starts with <var>scopeString</var> return <var>worker</var>.</li>
+        </ol>
+      </li>
+      <li>Return null.</li>
+    </ol>
+    </spec-algorithm>
+  </spec-section>
+
+  <spec-section id="on-foreign-fetch-request-algorithm">
+    <h1>Handle Foreign Fetch</h1>
```

Comment:
It would be nice if somehow some of the duplication between this and the regular fetch event handling could be shared. But maybe they will end up being too different for that not to matter much?